From f559efdd54b30766fbc32b742734d42d8274c4b6 Mon Sep 17 00:00:00 2001
From: Amrita H S <amrita.h.s@intel.com>
Date: Tue, 28 Aug 2018 11:22:12 +0530
Subject: [PATCH] Add ArtDefaultGCType as a soong variable.

This change helps us default the GenCopy GC as the default for broxton

Ported original gerrit link:
https://android.intel.com/#/c/621339/

Category: device enablement
Domain: AOSP.ART-Other
Origin: internal
Upstream-Candidate: no, proprietary
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-73559
Change-Id: Ideae554364da2dcde4554d252c91ddaccaa6bdb2
Signed-off-by: Amrita H S <amrita.h.s@intel.com>
---
 android/config.go   | 4 ++++
 android/variable.go | 1 +
 2 files changed, 5 insertions(+)

diff --git a/android/config.go b/android/config.go
index fafed6b..06169de 100644
--- a/android/config.go
+++ b/android/config.go
@@ -708,6 +708,10 @@ func (c *config) HostStaticBinaries() bool {
 	return Bool(c.productVariables.HostStaticBinaries)
 }
 
+func (c *config) ArtDefaultGCType() string {
+	return *c.productVariables.ArtDefaultGCType;
+}
+
 func (c *deviceConfig) Arches() []Arch {
 	var arches []Arch
 	for _, target := range c.config.Targets[Device] {
diff --git a/android/variable.go b/android/variable.go
index f4aaec7..666aac6 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -215,6 +215,7 @@ type productVariables struct {
 	SanitizeDeviceArch []string `json:",omitempty"`
 
 	ArtUseReadBarrier *bool `json:",omitempty"`
+	ArtDefaultGCType  *string `json:",omitempty"`
 
 	BtConfigIncludeDir *string `json:",omitempty"`
 
-- 
1.9.1

